nemo_aligner/utils/trt_llm.py

```diff
@@ -0,0 +1,175 @@
+import secrets
+
+import tensorrt_llm
+import torch
+
+from nemo.export.tensorrt_llm import TensorRTLLM
+from nemo.export.trt_llm import tensorrt_llm_run
+from nemo.export.trt_llm.nemo_ckpt_loader.nemo_file import build_tokenizer
+from nemo.utils import logging
+from nemo_aligner.utils import parallel_state
+from nemo_aligner.utils.distributed import broadcast_2d_tensor_within_mp
+
+
+class GPTGenerateTRTLLM:
+    def __init__(
+        self,
+        model_cfg,
+        max_generation_length=1024,
+        max_input_len=1024,
+        max_input_tokens=4096,
+        generation_batch_size=4,
+        unload_engine_train=False,
+        trt_model_type="GPTForCausalLM",
+        end_strings=None,
+        reshard_model=False,
+        sample_temperature=None,
+        sample_top_k=None,
+        sample_top_p=None,
+        repetition_penalty=1.0,
+        use_greedy=False,
+        tokenizer=None,
+        trt_model_dir="/tmp/trt_llm_model",
+        seed=None,
+    ):
+        if use_greedy and sample_top_k != 1:
+            if sample_top_k != 1:
+                logging.warning(f"'use_greedy=True' overrides {sample_top_k=} to 1")
+            sample_top_k = 1
+
+        self.model_cfg = model_cfg
+        self.tokenizer = tokenizer
+        self.max_generation_length = max_generation_length
+        self.max_input_len = max_input_len
+        self.max_input_tokens = max_input_tokens
+        self.generation_batch_size = generation_batch_size
+        self.unload_engine_train = unload_engine_train
+        self.trt_model_type = trt_model_type
+        self.reshard_model = reshard_model
+        self.pad_id = tokenizer.eos_id
+
+        self.trt_llm_exporter = TensorRTLLM(trt_model_dir, load_model=False)
+        self._trtllm_model_compiled = False
+
+        rng_generator = torch.Generator(device="cpu")
+        seed = secrets.randbits(32) if seed is None else seed
+
+        if seed is not None:
+            rng_generator.manual_seed(seed)
+
+        self.rng_generator = rng_generator
+
+        end_strings = list(end_strings)
+        end_strings = [[",".join(end_strings)] for _ in range(self.generation_batch_size)]
+        stop_list = tensorrt_llm_run.to_word_list_format(
+            end_strings, build_tokenizer(self.tokenizer), ref_str="green tea icecream"
```

Comment:
Can you add a comment regarding why `green tea icecream`? (I assume it's somewhat arbitrary, but better say so in order to avoid confusion)

Reply:
yup done! a2c7a679edf41a335664998e3aea135efd94f92a